lib/dispatcher/client-h2.js

```diff
@@ -278,6 +283,7 @@ function shouldSendContentLength (method) {
 }
 
 function writeH2 (client, request) {
+  const requestTimeout = request.bodyTimeout ?? client[kBodyTimeout]
```

Comment:
I was quite unsure if using the `bodyTimeout` for this, and adding a new one. But given the nature of the timeout, using the body's one should be enough.